client/cgo/verification.go

```diff
@@ -112,8 +142,15 @@ func cgoVerifyAuthPath(treeHash unsafe.Pointer, treeHashSize C.int,
 	ap.TreeNonce = tn
 	ap.LookupIndex = li
 	ap.PrunedTree = pt
+	ap.Leaf = &leafNode{
```

Comment:
> the client is going to be receiving messages that unmarshall like this anyways.

It's maybe a dumb question but do we still need to marshal the messages for transmissions?